subprojects/testing-native/src/test/groovy/org/gradle/nativeplatform/test/xctest/internal/XCTestSelectionTest.groovy

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.test.xctest.internal
+
+import org.gradle.testing.internal.util.Specification
+import spock.lang.Subject
+
+@Subject(XCTestSelection)
+class XCTestSelectionTest extends Specification {
+    def "includes all tests when no filter provided"() {
+        expect:
+        select([]).includedTests == [XCTestSelection.INCLUDE_ALL_TESTS]
+    }
+
+    def "throws IllegalArgumentException when filter has more than two dots"() {
+        when:
+        select(['more.than.two.dots'])
+
+        then:
+        def ex = thrown(IllegalArgumentException)
+        ex.message == 'Invalid pattern'
+    }
+
+    def "can use filters with two dots or fewer"() {
+        when:
+        select(['one.dot', 'has.two.dots'])
```

Comment:
What about `select(['ModuleName'])`?  I think this should be invalid too.

Reply:
It actually pass-through and will most likely not do anything as you would expect when using the command line. I didn't want to obfuscate too much the normal behavior of XCTest and not over assert so that we reimplement the test selection from XCTest. Do you think it's a good idea? I will add a test case for this.